File: pdfreporter-xcode/PDFReporter-ios/native/org/oss/pdfreporter/text/format/fallback/FallbackFormatFactory.m
//
//  FallbackFormatFactory.m
//  PDFReporter-ios
//
//  Created by Kendra on 9/24/13.
//  Copyright (c) 2014 Open Software Solutions GmbH. All rights reserved.
//

#import "FallbackFormatFactory.h"
#import "org/oss/pdfreporter/registry/ApiRegistry.h"
#import "FallbackFormater.h"

@implementation OrgOssPdfreporterTextFormatFallbackFallbackFormatFactory

+(void)registerFactory
{
    [OrgOssPdfreporterRegistryApiRegistry register__WithOrgOssPdfreporterTextFormatFactoryIFormatFactory_FormatTypeEnum:OrgOssPdfreporterTextFormatFactoryIFormatFactory_FormatTypeEnum_STANDARD withOrgOssPdfreporterTextFormatFactoryIFormatFactory: [[OrgOssPdfreporterTextFormatFallbackFallbackFormatFactory alloc] init]];
}

- (id<OrgOssPdfreporterTextFormatIDateFormat>)newDateFormatWithNSString:(NSString *)pattern
                                                     withJavaUtilLocale:(JavaUtilLocale *)locale
                                                   withJavaUtilTimeZone:(JavaUtilTimeZone *)timezone
{
    return [[FallbackFormater alloc] initWithPattern:pattern];
}

- (id<OrgOssPdfreporterTextFormatINumberFormat>)newNumberFormatWithNSString:(NSString *)pattern
                                                         withJavaUtilLocale:(JavaUtilLocale *)locale
{
    return [[FallbackFormater alloc] initWithPattern:pattern];
}

- (id<OrgOssPdfreporterTextFormatIMessageFormat>)newMessageFormatWithNSString:(NSString *)pattern
                                                           withJavaUtilLocale:(JavaUtilLocale *)locale
{
    return [[FallbackFormater alloc] initWithPattern:pattern];
}

void OrgOssPdfreporterTextFormatFallbackFallbackFormatFactory_registerFactory()
{
    [OrgOssPdfreporterTextFormatFallbackFallbackFormatFactory registerFactory];
}
@end

File: pdfreporter-java/src/org/oss/pdfreporter/text/format/fallback/FallbackFormatFactory.java
/*******************************************************************************
 * Copyright (c) 2013 Open Software Solutions GmbH.
 * All rights reserved. This program and the accompanying materials
 * are made available under the terms of the GNU Lesser Public License v3.0
 * which accompanies this distribution, and is available at
 * http://www.gnu.org/licenses/lgpl-3.0.html
 * 
 * Contributors:
 *     Open Software Solutions GmbH - initial API and implementation
 ******************************************************************************/
package org.oss.pdfreporter.text.format.fallback;

import java.util.Locale;
import java.util.TimeZone;

import org.oss.pdfreporter.registry.ApiRegistry;
import org.oss.pdfreporter.text.format.IDateFormat;
import org.oss.pdfreporter.text.format.IMessageFormat;
import org.oss.pdfreporter.text.format.INumberFormat;
import org.oss.pdfreporter.text.format.factory.IFormatFactory;


public class FallbackFormatFactory implements IFormatFactory {

	public static void registerFactory() {
		ApiRegistry.register(FormatType.STANDARD, new FallbackFormatFactory());
	}

	@Override
	public IDateFormat newDateFormat(String pattern, Locale locale,
			TimeZone timezone) {
		return pattern==null ? new DateFormat() : new DateFormat(pattern);
	}

	@Override
	public INumberFormat newNumberFormat(String pattern, Locale locale) {
		return pattern==null ? new NumberFormat() : new NumberFormat(pattern);
	}

	@Override
	public IMessageFormat newMessageFormat(String pattern, Locale locale) {
		return new MessageFormat(pattern);
	}
	
	

}
